x-pack/platform/plugins/shared/onechat/server/routes/tools.ts

```diff
@@ -125,10 +121,12 @@ export function registerToolsRoutes({ router, getInternalServices, logger }: Rou
           request: {
             body: schema.object({
               id: schema.string(),
-              type: schema.oneOf([schema.literal(ToolType.esql)]),
+              // @ts-expect-error schema.oneOf expects at least one element, and `map` returns a list
+              type: schema.oneOf(editableToolTypes.map((type) => schema.literal(type))),
               description: schema.string({ defaultValue: '' }),
               tags: schema.arrayOf(schema.string(), { defaultValue: [] }),
-              configuration: esqlConfigSchema,
+              // actual config validation is done in the tool service
+              configuration: schema.recordOf(schema.string(), schema.any()),
```

Comment:
Using the per-type schema here was a worse user experience because the config-schema/joi error messages for composite types are just atrocious. Fully delegating the per-type config validation to the tool service allows to only check against the type's schema, for better errors (and less complexity)